google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java

```diff
@@ -341,7 +341,8 @@ public void onSuccess(List<WriteResult> writeResults) {
                           }
                         });
                   }
-                });
+                },
+                userCallbackExecutor);
```

Comment:
Do we have any other `Executor` available in this class? If yes, then add test case or else we are good to go